prov/sockets/src/sock_ep_msg.c

```diff
@@ -702,13 +823,27 @@ static int sock_ep_cm_accept(struct fid_ep *ep, const void *param, size_t paraml
 	struct sockaddr_in *addr;
 	struct sock_ep *_ep;
 	struct sock_eq *_eq;
-	int ret;
+	int ret = 0;
 
 	_ep = container_of(ep, struct sock_ep, ep);
 	_eq = _ep->eq;
 	if (!_eq || paramlen > SOCK_EP_MAX_CM_DATA_SZ) 
 		return -FI_EINVAL;
 
+	if (_ep->is_disabled || _ep->cm.shutdown_received)
+		return -FI_EINVAL;
+
+	if (!_ep->cm.listener_thread) {
+		if (pthread_create(&_ep->cm.listener_thread, NULL, 
+				   sock_msg_ep_listener_thread, (void *)_ep)) {
+			SOCK_LOG_ERROR("Couldn't create listener thread\n");
+			return -FI_EINVAL;
+		}
+		
+		while (!(volatile int)_ep->cm.do_listen)
+			pthread_yield();
```

Comment:
Same here.  We really don't want to be allocating a new thread every time the user calls connect or accept.